src/coreclr/jit/codegencommon.cpp

```diff
@@ -3386,7 +3374,8 @@ void CodeGen::genHomeRegisterParams(regNumber initReg, bool* initRegStillZeroed)
             }
         }
 
-        if (compiler->info.compPublishStubParam && ((paramRegs & RBM_SECRET_STUB_PARAM) != RBM_NONE))
+        if (compiler->info.compPublishStubParam && ((paramRegs & RBM_SECRET_STUB_PARAM) != RBM_NONE) &&
+            compiler->lvaGetDesc(compiler->lvaStubArgumentVar)->lvOnFrame)
```

Comment:
This change was required because of the unification of the code in LSRA that figures out which registers are live-in. That code marks all registers assigned to untracked as live-in in `regState.rsCalleeRegArgMaskLiveIn`; however, even untracked locals may not have a stack home allocated if they have 0 ref count. This code would then try to store the register to its (non-existent) home.
We should update LSRA to avoid marking registers live-in in this case, but it comes with diffs (should be improvements), so I'd like to do it separately.